chat-plugins/blackjack.js

```diff
@@ -0,0 +1,501 @@
+		if (isNaN(target) || target <= 0) target = '';
+		if (target !== '') {
+			this.autostart = setTimeout(() => this.start(), (target * 60000));
+		}
+		if (!this.room.blackjack) this.room.blackjack = 1;
+		this.blackjackNumber = this.room.blackjack;
+
+		this.add(Chat.html`${this.createdBy} has created a game of Blackjack. ${(target !== '' ? `(automatically starting in ${target} minute${Chat.plural(target)}.)` : '')} <button class="button" name="send" value="/joingame">Join</button><br />`, null, true);
+	}
+	makePlayer(user) {
+		return new BlackjackPlayer(user, this);
+	}
+
+	/**
+	 * Joining/Leaving/Viewing
+	 * joinGame - joins the game
+	 * leaveGame - leaves the game
+	 * spectate - spectates the game
+	 * unspectate - stops spectating the game
+	 */
+	joinGame(user) {
+		if (!user.named) return this.errorMessage(user, `You must first choose a name to play Blackjack.`);
+		if (this.started) return this.errorMessage(user, `Blackjack has already started.`);
+		const joined = this.addPlayer(user, this.players);
+		if (!joined) {
+			this.errorMessage(user, `You are already in this game.`);
+			return false;
+		}
+		const comma = (this.atLeastOneJoin ? ', ' : '');
+		this.add(Chat.html`${comma}${user.name} joined`);
+
+		if (Object.keys(this.players).length === this.playerCap) {
+			this.start();
+		}
+		if (this.spectators[user]) delete this.spectators[user]; // prevent player from spectating
+		this.atLeastOneJoin = true;
+		return true;
+	}
+	leaveGame(user) {
+		if (this.started) return this.errorMessage(user, `You cannot leave this game; it has already started.`);
+		const left = this.removePlayer(user);
+		if (!left) return this.errorMessage(user, `You are not in this game to leave it.`);
+
+		this.add(Chat.html`, ${user.name} left`);
+	}
+	spectate(user) {
+		if (this.spectators[user]) return this.errorMessage(user, `You are already spectating this game.`);
+		if (this.players[user]) return this.errorMessage(user, `You don't need to spectate the game; you're playing the game.`);
+		this.spectators[user.userid] = user.userid;
+		user.sendTo(this.id, `You are now spectating this game.`);
+	}
+	unspectate(user) {
+		if (!this.spectators[user.userid]) return this.errorMessage(user, `You are already not spectating this game.`);
+		delete this.spectators[user.userid];
+		user.sendTo(this.id, `You are no longer spectating this game.`);
+	}
+
+	/**
+	 * Utility
+	 * errorMessage - sends a user an error message
+	 * add - adds/sends text to room
+	 * display - displays gameplay to players and spectators
+	 * clear - clears a user's gameplay screen
+	 */
+	errorMessage(user, message) {
+		user.sendTo(this.room, Chat.html`|html|<div class="message-error">${message}</div>`);
+	}
+	add(message, clean, add, endingMessage) {
+		const change = this.uhtmlChange;
+		if (!add) this.room.send(`|uhtml${change}|blackjack-${this.blackjackNumber}|<div class="infobox">${(clean ? message : this.lastMessage + message)}</div>`);
```

Comment:
This could be an if/else. Is `ChatRoom#update` only being called if `add` is truthy intentional?